File: decision_tree/elixir/dt.exs
defmodule DecisionTree do
  def run do
    # Load the dataset (Iris dataset as an example)
    iris = Sklearn.Datasets.load_iris()
    X = iris.data
    y = iris.target

    # Split the dataset into training and testing sets
    {X_train, X_test, y_train, y_test} = Sklearn.ModelSelection.train_test_split(X, y, test_size: 0.2, random_state: 42)

    # Create a decision tree classifier
    classifier = Sklearn.Tree.DecisionTreeClassifier.new()

    # Train the classifier on the training data
    classifier = Sklearn.Tree.DecisionTreeClassifier.fit(classifier, X_train, y_train)

    # Make predictions on the test data
    y_pred = Sklearn.Tree.DecisionTreeClassifier.predict(classifier, X_test)

    # Calculate the accuracy of the model
    accuracy = Sklearn.Metrics.accuracy_score(y_test, y_pred)
    IO.puts("Accuracy: #{accuracy}")
  end
end

DecisionTree.run()

File: decision_tree/cs/dt.cs
using System;
using System.Linq;
using Sklearn;
using Sklearn.ModelSelection;
using Sklearn.Tree;
using Sklearn.Metrics;

class Program
{
    static void Main(string[] args)
    {
        // Load the dataset (Iris dataset as an example)
        var iris = Datasets.LoadIris();
        var X = iris.Data;
        var y = iris.Target;

        // Split the dataset into training and testing sets
        var random = new Random(42);
        var splitter = new TrainTestSplit(randomState: random);
        var (X_train, X_test, y_train, y_test) = splitter.Split(X, y, test_size: 0.2);

        // Create a decision tree classifier
        var classifier = new DecisionTreeClassifier();

        // Train the classifier on the training data
        classifier.Fit(X_train, y_train);

        // Make predictions on the test data
        var y_pred = classifier.Predict(X_test);

        // Calculate the accuracy of the model
        var accuracy = Metrics.AccuracyScore(y_test, y_pred);
        Console.WriteLine("Accuracy: " + accuracy);
    }
}
